src/wmtk/simplex/link_condition.cpp

```diff
@@ -0,0 +1,90 @@
+#include "link_condition.hpp"
+#include "link.hpp"
+#include "open_star.hpp"
+
+namespace wmtk::simplex {
+bool link_condition_closed_trimesh(const TriMesh& mesh, const Tuple& edge)
+{
+    SimplexCollection lnk_a = link(mesh, Simplex(PrimitiveType::Vertex, edge), true); // link(a)
+    SimplexCollection lnk_b = link(
+        mesh,
+        Simplex(PrimitiveType::Vertex, mesh.switch_tuple(edge, PrimitiveType::Vertex)),
+        true); // link(b)
+    SimplexCollection lnk_ab = link(mesh, Simplex(PrimitiveType::Edge, edge), true); // link(ab)
+
+    SimplexCollection lnk_a_lnk_b_intersection = SimplexCollection::get_intersection(lnk_a, lnk_b);
+
+    return SimplexCollection::are_simplex_collections_equal(lnk_a_lnk_b_intersection, lnk_ab);
+}
+
+bool link_condition(const EdgeMesh& mesh, const Tuple& edge)
+{
+    const Tuple edge_switch_v = mesh.switch_tuple(edge, PrimitiveType::Vertex);
+    if (mesh.is_boundary_vertex(edge) && mesh.is_boundary_vertex(edge_switch_v)) {
+        return false;
+    }
+    if (mesh.simplices_are_equal(
+            Simplex(PrimitiveType::Vertex, edge),
+            Simplex(PrimitiveType::Vertex, edge_switch_v))) {
+        return false;
+    }
+    return true;
+}
+
+bool link_condition(const TriMesh& mesh, const Tuple& edge)
+{
+    // step1 check link condition for closed case
+    if (!link_condition_closed_trimesh(mesh, edge)) {
+        return false;
+    }
+
+    // check if dummy vertex w is included in the lhs
```

Comment:
What is "dummy vertex w"?

Reply:
It’s a notation in this paper http://elib.mi.sanu.ac.rs/files/journals/publ/86/n080p023.pdf

Reply:
I see. However, the code should be understandable even without reading a paper. You can still add a hint in the documentation of the method that this function implements the paper but reading the paper should not be required to understand the code.